### PR TITLE
Merge f/notebook-controller into main

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Wrap up the JH dependencies removal work.

JIRA: https://issues.redhat.com/browse/RHOAIENG-572

### CODE REVIEW

**Comment by andrewballantyne** [comment]
/hold

We should merge this next sprint asap -- this is too risky to do this sprint.

**Comment by Gkrumbach07** [comment]
@andrewballantyne Is this good to be merged in this sprint?

**Comment by andrewballantyne** [comment]
Let us merge this as soon as we are confident we have a solid solution here.

**Comment by zdtsw** [comment]
do you plan to get this one out this week? we have a odh 2.9 code freeze planned on 8th March @andrewballantyne

**Comment by zdtsw** [comment]
can we get this in ODH 2.9 (tomorrow) code freeze?

**Comment by andrewballantyne** [comment]
@Gkrumbach07 Do you have time to test this out?

**Comment by Gkrumbach07** [comment]
is this part of the feature?

https://issues.redhat.com/browse/RHOAIENG-573

it was linked in the jira

File: frontend/src/pages/projects/screens/projects/ProjectListView.tsx [hunk 1]
```diff
@@ -97,11 +95,9 @@ const ProjectListView: React.FC<ProjectListViewProps> = ({ allowCreate, scope })
                 />
               </ToolbarItem>
             )}
-            {dashboardConfig.spec.notebookController?.enabled && (
```
Comment (Gkrumbach07) [inline frontend/src/pages/projects/screens/projects/ProjectListView.tsx:100]: why was this removed. should we check still

File: docs/dashboard-config.md [hunk 2]
```diff
@@ -150,6 +150,9 @@ spec:
     disablePerformanceMetrics: false
   notebookController:
     enabled: true
+    gpuSetting: autodetect
+    pvcSize: 20Gi
+    notebookNamespace: odh-notebooks
```
Comment (Gkrumbach07) [inline docs/dashboard-config.md:155]: remomved from docs but is still used in the code?

**Comment by Gkrumbach07** [comment]
Screenshot 2024-03-07 at 2 37 17 PM

i am getting two when it is enabled? not sure why the old jupyterhub wont go away

File: backend/src/routes/api/components/list.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ export const listComponents = async (
   request: FastifyRequest,
 ): Promise<OdhApplication[]> => {
   const applications = getApplications().filter(
-    (component) => component.metadata.name !== (checkJupyterEnabled() ? 'jupyterhub' : 'jupyter'),
```
Comment (Gkrumbach07) [inline backend/src/routes/api/components/list.ts:11]: if they still have jupyterhub as a app, then this never goes away

File: docs/dashboard-config.md (hunk 2 above)
Comment (Gkrumbach07) [reply docs/dashboard-config.md:155]: specifically `notebookNamespace`

**Comment by Gkrumbach07** [comment]
there are still references to `jupyterhub` in some manifests but they seem not to impact anything, thoughts @DaoDaoNoCode?

**Comment by andrewballantyne** [comment]
> there are still references to `jupyterhub` in some manifests but they seem not to impact anything, thoughts @DaoDaoNoCode?

We shouldn't have any of those -- right @DaoDaoNoCode ?

**Comment by DaoDaoNoCode** [comment]
> We shouldn't have any of those -- right @DaoDaoNoCode ?

No we shouldn't, let me double check.

**Comment by Gkrumbach07** [comment]
`AppLauncher` component has a stack trace error
Screenshot 2024-03-07 at 3 07 40 PM

**Comment by Gkrumbach07** [comment]
> I didn't see it anywhere from the manifest.

Screenshot 2024-03-07 at 3 08 54 PM

this is what im seeing

File: backend/src/routes/api/components/list.ts (hunk 3 above)
Comment (Gkrumbach07) [reply backend/src/routes/api/components/list.ts:11]: this will be resolved by another pr that cleans up the old app

File: docs/dashboard-config.md (hunk 2 above)
Comment (Gkrumbach07) [reply docs/dashboard-config.md:155]: nvm i think i looked at this wrong

**Comment by Gkrumbach07** [comment]
going to test one last time after the rebase

**Comment by Gkrumbach07** [comment]
/approve
rebase fixed the issue with AppLauncher component has a stack trace error
